Refactor HomePage's script

colewinfield opened the pull request:
Moved the HomePage's JavaScript functions from the site.js to its own file, then referenced the script in the Razor HTML file. Better SOC this way, as suggested by @rlayman87.

colewinfield committed:
Refactor HomePage's script

Moved the HomePage's JavaScript functions from the site.js to its own file, then referenced the script in the Razor HTML file. Better SOC this way.

jonnywwright commented on src/Mentat.UI/wwwroot/js/Home.js:
```diff
@@ -0,0 +1,22 @@
+            para[j].innerHTML += "...";
+        }
+    }
+}
```
Can this same thing be accomplished with css using something like line-clamp?
https://css-tricks.com/almanac/properties/l/line-clamp/

colewinfield replied:
You'd think I would have been able to find that property, or at least have it mentioned on SO, when I was searching the other night. That's a much easier way to implement it, thanks, @jonnywwright,

jonnywwright replied:
I hear ya. Here is the css code I tested.

```css
.long-title {
    display: -webkit-box;
    -webkit-line-clamp: 2;
    -webkit-box-orient: vertical;
    overflow: hidden;
}

.long-text {
    display: -webkit-box;
    -webkit-line-clamp: 3;
    -webkit-box-orient: vertical;
    overflow: hidden;
}
```

![image](https://user-images.githubusercontent.com/20952808/179022496-506f1846-0735-4b73-b97a-4c225f8cfe16.png)

jonnywwright replied:
The biggest benefit will eventually be, whenever the design is figured out the text will collapse with the window resizing. I think we're a ways off from this however. Edit -- I just saw your commit and looks like you got it. Well done. Don't forget to delete the js function.

colewinfield committed:
Use line-clamp property

Instead of the JS function, use the line-clamp property to truncate lines of text.

rlayman87 reviewed (approved):
Re-approving.

jonnywwright commented:
@rlayman87 you merged this code on behalf of @colewinfield while my comment was unresolved and the requested code change was in-complete. The js function needs to be removed as it has been replaced with css. Please be more diligent when merging others work into master and development.

cc @dyohn

colewinfield commented:
@jonnywwright That was my fault, I re-requested @rlayman87's approval but absentmindedly forgot to delete the Home.js file. It's not a huge deal.

rlayman87 commented:
> @rlayman87 you merged this code on behalf of @colewinfield while my comment was unresolved and the requested code change was in-complete. The js function needs to be removed as it has been replaced with css. Please be more diligent when merging others work into master and development.
> 
> cc @dyohn

Hi, @jonnywwright
You had approved this change. If you were waiting on changes, you should have used the "Comment" or "Request changes" option
![image](https://user-images.githubusercontent.com/55672598/179235526-410d147f-9873-4b44-82a0-aff70d9cc0ab.png)
instead of "Approve", so it would have been clear to everyone that you were waiting on changes. But ultimately, I do agree with @colewinfield, it's not a big deal and he can just make that change with his next commit and merge he is working on.
Please let me know if you have any additional questions or comments.
Thank you!
Rachel

jonnywwright commented:
> > @rlayman87 you merged this code on behalf of @colewinfield while my comment was unresolved and the requested code change was in-complete. The js function needs to be removed as it has been replaced with css. Please be more diligent when merging others work into master and development.
> > cc @dyohn
> 
> Hi, @jonnywwright You had approved this change. If you were waiting on changes, you should have used the "Comment" or "Request changes" option ![image](https://user-images.githubusercontent.com/55672598/179235526-410d147f-9873-4b44-82a0-aff70d9cc0ab.png) instead of "Approve", so it would have been clear to everyone that you were waiting on changes. But ultimately, I do agree with @colewinfield, it's not a big deal and he can just make that change with his next commit and merge he is working on. Please let me know if you have any additional questions or comments. Thank you! Rachel

Ah I see, and thanks for showing me that. I will be sure to use Request Changes to help prevent these mistakes in the future.

cc @dyohn

rlayman87 commented:
> > > @rlayman87 you merged this code on behalf of @colewinfield while my comment was unresolved and the requested code change was in-complete. The js function needs to be removed as it has been replaced with css. Please be more diligent when merging others work into master and development.
> > > cc @dyohn
> > 
> > 
> > Hi, @jonnywwright You had approved this change. If you were waiting on changes, you should have used the "Comment" or "Request changes" option ![image](https://user-images.githubusercontent.com/55672598/179235526-410d147f-9873-4b44-82a0-aff70d9cc0ab.png) instead of "Approve", so it would have been clear to everyone that you were waiting on changes. But ultimately, I do agree with @colewinfield, it's not a big deal and he can just make that change with his next commit and merge he is working on. Please let me know if you have any additional questions or comments. Thank you! Rachel
> 
> Ah I see, and thanks for showing me that. I will be sure to use Request Changes to help prevent these mistakes in the future.
> 
> cc @dyohn

@jonnywwright not a problem.
